File: ws/palomo/u2/ws18GUISwing/ShoppingCenter_Python/customer.py
"""
customer.py
Modelo de cliente para el Shopping Center
@author Cristian (convertido a Python)
"""


class Customer:
    def __init__(self, id: int, first_name: str, last_name: str,
                 gender: str, money_spent: float, age: int,
                 type_customer: str, hobbies: list = None):
        self.__id = id
        self.__first_name = first_name
        self.__last_name = last_name
        self.__gender = gender
        self.__money_spent = money_spent
        self.__age = age
        self.__type_customer = type_customer
        self.__hobbies = hobbies if hobbies is not None else []

    # --- Getters ---
    def get_id(self) -> int:
        return self.__id

    def get_first_name(self) -> str:
        return self.__first_name

    def get_last_name(self) -> str:
        return self.__last_name

    def get_gender(self) -> str:
        return self.__gender

    def get_money_spent(self) -> float:
        return self.__money_spent

    def get_age(self) -> int:
        return self.__age

    def get_type_customer(self) -> str:
        return self.__type_customer

    def get_hobbies(self) -> list:
        return self.__hobbies

    # --- Setters ---
    def set_id(self, id: int):
        self.__id = id

    def set_first_name(self, first_name: str):
        self.__first_name = first_name

    def set_last_name(self, last_name: str):
        self.__last_name = last_name

    def set_gender(self, gender: str):
        self.__gender = gender

    def set_money_spent(self, money_spent: float):
        self.__money_spent = money_spent

    def set_age(self, age: int):
        self.__age = age

    def set_type_customer(self, type_customer: str):
        self.__type_customer = type_customer

    def set_hobbies(self, hobbies: list):
        self.__hobbies = hobbies

    def __str__(self):
        return (f"Customer {{ id={self.__id}, firstName={self.__first_name}, "
                f"lastName={self.__last_name}, gender={self.__gender}, "
                f"moneySpent={self.__money_spent}, age={self.__age}, "
                f"typeCustomer={self.__type_customer}, hobbies={self.__hobbies} }}")

File: hw/palomo/u2/hw13GUISwing/ShoppingCenter_JS/src/Customer.js
/**
 * Customer.js
 * Modelo de cliente para el Shopping Center
 * @author Cristian (convertido a JavaScript)
 */
class Customer {
    /**
     * @param {number} id
     * @param {string} firstName
     * @param {string} lastName
     * @param {string} gender
     * @param {number} moneySpent
     * @param {number} age
     * @param {string} typeCustomer
     * @param {string[]} hobbies
     */
    constructor(id, firstName, lastName, gender, moneySpent, age, typeCustomer, hobbies = []) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.gender = gender;
        this.moneySpent = moneySpent;
        this.age = age;
        this.typeCustomer = typeCustomer;
        this.hobbies = hobbies;
    }

    // Getters
    getId() { return this.id; }
    getFirstName() { return this.firstName; }
    getLastName() { return this.lastName; }
    getGender() { return this.gender; }
    getMoneySpent() { return this.moneySpent; }
    getAge() { return this.age; }
    getTypeCustomer() { return this.typeCustomer; }
    getHobbies() { return this.hobbies; }

    // Setters
    setId(id) { this.id = id; }
    setFirstName(firstName) { this.firstName = firstName; }
    setLastName(lastName) { this.lastName = lastName; }
    setGender(gender) { this.gender = gender; }
    setMoneySpent(moneySpent) { this.moneySpent = moneySpent; }
    setAge(age) { this.age = age; }
    setTypeCustomer(typeCustomer) { this.typeCustomer = typeCustomer; }
    setHobbies(hobbies) { this.hobbies = hobbies; }

    toString() {
        return `Customer { id=${this.id}, firstName=${this.firstName}, lastName=${this.lastName}, ` +
               `gender=${this.gender}, moneySpent=${this.moneySpent}, age=${this.age}, ` +
               `typeCustomer=${this.typeCustomer}, hobbies=[${this.hobbies.join(', ')}] }`;
    }
}

module.exports = Customer;
